### DISCUSSION TITLE
Library for HiLetgo 5pcs Capacitive Soil Moisture Sensor v 2.0

### DISCUSSION THREAD

**Discussion by Ricmelis80** [micropython/micropython#12542] [Core Development]
Hy everyone! 

I'll tryng to transfer soil moisture data from my Raspberry Pico W to my PC.  I used in this sensor model 
https://www.amazon.it/HiLetgo-Capacitive-Moisture-Voltage-3-3-5-5V/dp/B07WCB3GCB/ref=sr_1_1_sspa?camp=1789&creative=9325&keywords=capacitive+soil+moisture+sensor&linkCode=gs3&linkId=2f85a9b72865575990a06eb0c5f3ad75&qid=1695884408&sr=8-1-spons&sp_csd=d2lkZ2V0TmFtZT1zcF9hdGY&psc=1) 

Unfortunately, I dont find any Micropython library for this specific sensor model... How I can resolve it? Many thanks in advances!

**Comment by robert-hh** [comment]
From the schematics it looks like you only have to read the analog voltage at the output. To read a value, look at the documentation for the machine.ADC() class. e.g. here: https://docs.micropython.org/en/latest/rp2/quickref.html#adc-analog-to-digital-conversion

**Reply by Ricmelis80** [reply]
Hy Robert.

Thanks a lot!

**Reply by 2dof** [reply]
remember to perform calibration (best do measurements couple times and average values ) according to [data sheet](https://www.sigmaelectronica.net/wp-content/uploads/2018/04/sen0193-humedad-de-suelos.pdf) .
